From 3bc99711d81a2015fb0b64abec46e67c0e6f0069 Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Fri, 13 May 2022 12:21:55 +0530
Subject: [PATCH 20/30] Launcher3: Add un-themed icons to google search bar

Themed icons should only be used when themed icons are
enabled by user in ThemePicker. Otherwise, use the
standard google colors. Matches Pixel Launcher.

Change-Id: If6e81f746f2ab7875da81fd24b3663fad03e5fe9
Co-authored-by: Pranav Vashi <neobuddy89@gmail.com>
Co-authored-by: Patryk Michalik <contact@patrykmichalik.com>
Signed-off-by: Adithya R <gh0strider.2k18.reborn@gmail.com>

[YAAP] update for U
---
 res/drawable/ic_lens_color.xml                | 30 +++++-----
 res/drawable/ic_lens_themed.xml               | 27 +++++++++
 res/drawable/ic_mic_color.xml                 |  4 +-
 res/drawable/ic_mic_themed.xml                | 23 ++++++++
 res/drawable/ic_super_g_themed.xml            | 23 ++++++++
 .../launcher3/qsb/AssistantIconView.java      |  6 +-
 src/com/android/launcher3/qsb/QsbLayout.java  | 56 ++++++++++++++-----
 7 files changed, 136 insertions(+), 33 deletions(-)
 create mode 100644 res/drawable/ic_lens_themed.xml
 create mode 100644 res/drawable/ic_mic_themed.xml
 create mode 100644 res/drawable/ic_super_g_themed.xml

diff --git a/res/drawable/ic_lens_color.xml b/res/drawable/ic_lens_color.xml
index 3f1b8e98e7..f8eea294cf 100644
--- a/res/drawable/ic_lens_color.xml
+++ b/res/drawable/ic_lens_color.xml
@@ -1,22 +1,26 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:theme="@style/QsbIconTint"
-    android:height="24dp"
-    android:width="24dp"
-    android:viewportWidth="192"
-    android:viewportHeight="192">
+    android:width="32dp"
+    android:height="32dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
     <path
-        android:fillColor="#4285F4"
-        android:pathData="M68,104.15 A28,28 0 1,1 124,104.15 A28,28 0 1,1 68,104.15" />
+        android:name="qsbIconTintTertiary"
+        android:fillColor="#FBBC05"
+        android:pathData="M16,5C17.66,5 19,6.34 19,8V10H21V8C21,5.24 18.76,3 16,3H14V5H16Z" />
     <path
-        android:fillColor="#EA4335"
-        android:pathData="M160,72v40.15V136c0,1.69-0.34,3.29-0.82,4.82v0v0c-1.57,4.92-5.43,8.78-10.35,10.35h0v0 c-1.53,0.49-3.13,0.82-4.82,0.82H66l16,16h50h12c4.42,0,8.63-0.9,12.46-2.51c3.83-1.62,7.28-3.96,10.17-6.86 c1.45-1.45,2.76-3.03,3.91-4.74c2.3-3.4,3.96-7.28,4.81-11.44c0.43-2.08,0.65-4.24,0.65-6.45v-12V96.15V84l-6-19l-10.82,2.18 C159.66,68.71,160,70.31,160,72z" />
+        android:name="qsbIconTintPrimary"
+        android:fillColor="#4285F4"
+        android:pathData="M8,19C6.34,19 5,17.66 5,16V14H3V16C3,18.76 5.24,21 8,21H10V19H8Z" />
     <path
+        android:name="qsbIconTintPrimary2"
         android:fillColor="#4285F4"
-        android:pathData="M32,72c0-1.69,0.34-3.29,0.82-4.82c1.57-4.92,5.43-8.78,10.35-10.35C44.71,56.34,46.31,56,48,56 h96c1.69,0,3.29,0.34,4.82,0.82c0,0,0,0,0,0L149,45l-17-5l-16-16h-13.44H96h-6.56H76L60,40H48c-17.67,0-32,14.33-32,32v12v20l16,16 V72z" />
+        android:pathData="M8.5,12C8.4976,11.2595 8.7304,10.537 9.1645,9.937C9.5986,9.3371 10.2123,8.8901 10.9164,8.6609C11.6205,8.4316 12.3795,8.4316 13.0836,8.6609C13.7878,8.8901 14.4014,9.3371 14.8355,9.937C15.2696,10.537 15.5024,11.2595 15.5,12C15.5024,12.7405 15.2696,13.463 14.8355,14.063C14.4014,14.6629 13.7878,15.1099 13.0836,15.3391C12.3795,15.5684 11.6205,15.5684 10.9164,15.3391C10.2123,15.1099 9.5986,14.6629 9.1645,14.063C8.7304,13.463 8.4976,12.7405 8.5,12Z" />
     <path
+        android:name="qsbIconTintSecondary"
         android:fillColor="#34A853"
-        android:pathData="M144,40h-12l16.83,16.83c1.23,0.39,2.39,0.93,3.47,1.59c2.16,1.32,3.97,3.13,5.29,5.29 c0.66,1.08,1.2,2.24,1.59,3.47v0L176,84V72C176,54.33,161.67,40,144,40z"/>
+        android:pathData="M16,18C16,17.4697 16.2109,16.9608 16.5858,16.5858C16.9608,16.2109 17.4697,16 18,16C18.5303,16 19.0392,16.2109 19.4142,16.5858C19.7891,16.9608 20,17.4697 20,18C20,18.5303 19.7891,19.0392 19.4142,19.4142C19.0392,19.7891 18.5303,20 18,20C17.4697,20 16.9608,19.7891 16.5858,19.4142C16.2109,19.0392 16,18.5303 16,18Z" />
     <path
-        android:fillColor="#FBBC05"
-        android:pathData="M48,168h39.89l-16-16H48c-8.82,0-16-7.18-16-16v-23.89l-16-16V136C16,153.67,30.33,168,48,168z"/>
+        android:name="qsbIconTintQuaternary"
+        android:fillColor="#EA4335"
+        android:pathData="M8,5C6.34,5 5,6.34 5,8V10H3V8C3,5.24 5.24,3 8,3H10V5H8Z" />
 </vector>
diff --git a/res/drawable/ic_lens_themed.xml b/res/drawable/ic_lens_themed.xml
new file mode 100644
index 0000000000..892ab4360b
--- /dev/null
+++ b/res/drawable/ic_lens_themed.xml
@@ -0,0 +1,27 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="32dp"
+    android:height="32dp"
+    android:theme="@style/QsbIconTint"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path
+        android:name="qsbIconTintTertiary"
+        android:fillColor="?qsbIconTintTertiary"
+        android:pathData="M16,5C17.66,5 19,6.34 19,8V10H21V8C21,5.24 18.76,3 16,3H14V5H16Z" />
+    <path
+        android:name="qsbIconTintPrimary"
+        android:fillColor="?qsbIconTintPrimary"
+        android:pathData="M8,19C6.34,19 5,17.66 5,16V14H3V16C3,18.76 5.24,21 8,21H10V19H8Z" />
+    <path
+        android:name="qsbIconTintPrimary2"
+        android:fillColor="?qsbIconTintPrimary"
+        android:pathData="M8.5,12C8.4976,11.2595 8.7304,10.537 9.1645,9.937C9.5986,9.3371 10.2123,8.8901 10.9164,8.6609C11.6205,8.4316 12.3795,8.4316 13.0836,8.6609C13.7878,8.8901 14.4014,9.3371 14.8355,9.937C15.2696,10.537 15.5024,11.2595 15.5,12C15.5024,12.7405 15.2696,13.463 14.8355,14.063C14.4014,14.6629 13.7878,15.1099 13.0836,15.3391C12.3795,15.5684 11.6205,15.5684 10.9164,15.3391C10.2123,15.1099 9.5986,14.6629 9.1645,14.063C8.7304,13.463 8.4976,12.7405 8.5,12Z" />
+    <path
+        android:name="qsbIconTintSecondary"
+        android:fillColor="?qsbIconTintSecondary"
+        android:pathData="M16,18C16,17.4697 16.2109,16.9608 16.5858,16.5858C16.9608,16.2109 17.4697,16 18,16C18.5303,16 19.0392,16.2109 19.4142,16.5858C19.7891,16.9608 20,17.4697 20,18C20,18.5303 19.7891,19.0392 19.4142,19.4142C19.0392,19.7891 18.5303,20 18,20C17.4697,20 16.9608,19.7891 16.5858,19.4142C16.2109,19.0392 16,18.5303 16,18Z" />
+    <path
+        android:name="qsbIconTintQuaternary"
+        android:fillColor="?qsbIconTintQuaternary"
+        android:pathData="M8,5C6.34,5 5,6.34 5,8V10H3V8C3,5.24 5.24,3 8,3H10V5H8Z" />
+</vector>
diff --git a/res/drawable/ic_mic_color.xml b/res/drawable/ic_mic_color.xml
index 7c00274684..3c54cf1d29 100644
--- a/res/drawable/ic_mic_color.xml
+++ b/res/drawable/ic_mic_color.xml
@@ -1,6 +1,6 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="24dp"
-    android:height="24dp"
+    android:width="32dp"
+    android:height="32dp"
     android:viewportWidth="24"
     android:viewportHeight="24">
     <path
diff --git a/res/drawable/ic_mic_themed.xml b/res/drawable/ic_mic_themed.xml
new file mode 100644
index 0000000000..2e9565441e
--- /dev/null
+++ b/res/drawable/ic_mic_themed.xml
@@ -0,0 +1,23 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="32dp"
+    android:height="32dp"
+    android:theme="@style/QsbIconTint"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path
+        android:name="qsbIconTintPrimary"
+        android:fillColor="?qsbIconTintPrimary"
+        android:pathData="M12,15c1.66,0 2.99,-1.34 2.99,-3L15,5c0,-1.66 -1.34,-3 -3,-3S9,3.34 9,5v7c0,1.66 1.34,3 3,3z" />
+    <path
+        android:name="qsbIconTintSecondary"
+        android:fillColor="?qsbIconTintSecondary"
+        android:pathData="M11,18.92h2V22h-2z" />
+    <path
+        android:name="qsbIconTintTertiary"
+        android:fillColor="?qsbIconTintTertiary"
+        android:pathData="M7,12H5c0,1.93 0.78,3.68 2.05,4.95l1.41,-1.41C7.56,14.63 7,13.38 7,12z" />
+    <path
+        android:name="qsbIconTintQuaternary"
+        android:fillColor="?qsbIconTintQuaternary"
+        android:pathData="M12,17c-1.38,0 -2.63,-0.56 -3.54,-1.47l-1.41,1.41C8.32,18.21 10.07,19 12.01,19c3.87,0 6.98,-3.14 6.98,-7h-2c0,2.76 -2.23,5 -4.99,5z" />
+</vector>
diff --git a/res/drawable/ic_super_g_themed.xml b/res/drawable/ic_super_g_themed.xml
new file mode 100644
index 0000000000..13abf8e604
--- /dev/null
+++ b/res/drawable/ic_super_g_themed.xml
@@ -0,0 +1,23 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:theme="@style/QsbIconTint"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:name="qsbIconTintPrimary"
+      android:pathData="M23.49,12.27C23.49,11.48 23.42,10.73 23.3,10H12V14.51H18.47C18.18,15.99 17.33,17.24 16.07,18.09V21.09H19.93C22.19,19 23.49,15.92 23.49,12.27Z"
+      android:fillColor="?qsbIconTintPrimary"/>
+  <path
+      android:name="qsbIconTintSecondary"
+      android:pathData="M12,24C15.24,24 17.95,22.92 19.93,21.09L16.07,18.09C14.99,18.81 13.62,19.25 12,19.25C8.87,19.25 6.22,17.14 5.27,14.29H1.29V17.38C3.26,21.3 7.31,24 12,24Z"
+      android:fillColor="?qsbIconTintSecondary"/>
+  <path
+      android:name="qsbIconTintTertiary"
+      android:pathData="M5.27,14.29C5.02,13.57 4.89,12.8 4.89,12C4.89,11.2 5.03,10.43 5.27,9.71V6.62H1.29C0.47,8.24 0,10.06 0,12C0,13.94 0.47,15.76 1.29,17.38L5.27,14.29Z"
+      android:fillColor="?qsbIconTintTertiary"/>
+  <path
+      android:name="qsbIconTintQuaternary"
+      android:pathData="M12,4.75C13.77,4.75 15.35,5.36 16.6,6.55L20.02,3.13C17.95,1.19 15.24,0 12,0C7.31,0 3.26,2.7 1.29,6.62L5.27,9.71C6.22,6.86 8.87,4.75 12,4.75Z"
+      android:fillColor="?qsbIconTintQuaternary"/>
+</vector>
diff --git a/src/com/android/launcher3/qsb/AssistantIconView.java b/src/com/android/launcher3/qsb/AssistantIconView.java
index 2513112222..0821164b8e 100644
--- a/src/com/android/launcher3/qsb/AssistantIconView.java
+++ b/src/com/android/launcher3/qsb/AssistantIconView.java
@@ -19,10 +19,6 @@ public class AssistantIconView extends ImageView {
         });
     }
 
-    public void setIcon() {
-        setImageResource(R.drawable.ic_mic_color);
-    }
-
     public AssistantIconView(Context context, AttributeSet attrs, int defStyle) {
         super(context, attrs, defStyle);
         setScaleType(ScaleType.CENTER);
@@ -32,4 +28,4 @@ public class AssistantIconView extends ImageView {
         });
     }
 
-}
\ No newline at end of file
+}
diff --git a/src/com/android/launcher3/qsb/QsbLayout.java b/src/com/android/launcher3/qsb/QsbLayout.java
index 17ae484ad8..ab45a5e8e1 100644
--- a/src/com/android/launcher3/qsb/QsbLayout.java
+++ b/src/com/android/launcher3/qsb/QsbLayout.java
@@ -1,26 +1,32 @@
 package com.android.launcher3.qsb;
 
+import android.app.Activity;
 import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.net.Uri;
 import android.os.Bundle;
 import android.os.SystemClock;
 import android.util.AttributeSet;
+import android.view.View;
 import android.widget.FrameLayout;
-import android.widget.ImageButton;
+import android.widget.ImageView;
 import androidx.core.view.ViewCompat;
 import com.android.launcher3.R;
 import com.android.launcher3.Reorderable;
 import com.android.launcher3.Utilities;
 import com.android.launcher3.qsb.QsbContainerView;
 import com.android.launcher3.util.MultiTranslateDelegate;
-import android.view.View;
+import com.android.launcher3.util.Themes;
+import com.android.launcher3.LauncherPrefs;
 
-public class QsbLayout extends FrameLayout implements Reorderable {
+public class QsbLayout extends FrameLayout implements Reorderable,
+        SharedPreferences.OnSharedPreferenceChangeListener {
 
-    ImageButton lensIcon;
-    AssistantIconView assistantIcon;
+    ImageView mAssistantIcon;
+    ImageView mGoogleIcon;
+    ImageView mLensIcon;
     Context mContext;
 
     private final MultiTranslateDelegate mTranslateDelegate = new MultiTranslateDelegate(this);
@@ -35,18 +41,25 @@ public class QsbLayout extends FrameLayout implements Reorderable {
         super(context, attrs, defStyle);
         mContext = context;
     }
+
     @Override
     protected void onFinishInflate() {
         super.onFinishInflate();
-        assistantIcon = findViewById(R.id.mic_icon);
-        assistantIcon.setIcon();
+        mAssistantIcon = findViewById(R.id.mic_icon);
+        mGoogleIcon = findViewById(R.id.g_icon);
+        mLensIcon = findViewById(R.id.lens_icon);
+        setIcons();
+
+        LauncherPrefs.getPrefs(mContext).registerOnSharedPreferenceChangeListener(this);
+
         String searchPackage = QsbContainerView.getSearchWidgetPackageName(mContext);
         setOnClickListener(view -> {
             mContext.startActivity(new Intent("android.search.action.GLOBAL_SEARCH").addFlags(Intent.FLAG_ACTIVITY_NEW_TASK |
                 Intent.FLAG_ACTIVITY_CLEAR_TASK).setPackage(searchPackage));
         });
+
         if (Utilities.isGSAEnabled(mContext)) {
-            setupLensIcon();
+            enableLensIcon();
         }
     }
 
@@ -65,11 +78,28 @@ public class QsbLayout extends FrameLayout implements Reorderable {
         }
     }
 
-    private void setupLensIcon() {
-        lensIcon = findViewById(R.id.lens_icon);
-        lensIcon.setVisibility(View.VISIBLE);
-        lensIcon.setImageResource(R.drawable.ic_lens_color);
-        lensIcon.setOnClickListener(view -> {
+    @Override
+    public void onSharedPreferenceChanged(SharedPreferences prefs, String key) {
+        if (key.equals(Themes.KEY_THEMED_ICONS)) {
+            setIcons();
+        }
+    }
+
+    private void setIcons() {
+        if (Themes.isThemedIconEnabled(mContext)) {
+            mAssistantIcon.setImageResource(R.drawable.ic_mic_themed);
+            mGoogleIcon.setImageResource(R.drawable.ic_super_g_themed);
+            mLensIcon.setImageResource(R.drawable.ic_lens_themed);
+        } else {
+            mAssistantIcon.setImageResource(R.drawable.ic_mic_color);
+            mGoogleIcon.setImageResource(R.drawable.ic_super_g_color);
+            mLensIcon.setImageResource(R.drawable.ic_lens_color);
+        }
+    }
+
+    private void enableLensIcon() {
+        mLensIcon.setVisibility(View.VISIBLE);
+        mLensIcon.setOnClickListener(view -> {
             Intent lensIntent = new Intent();
             Bundle bundle = new Bundle();
             bundle.putString("caller_package", Utilities.GSA_PACKAGE);
-- 
2.52.0

